private/buf/bufgen/generator.go

```diff
@@ -29,8 +29,8 @@ import (
 )
 
 const (
-	// javaPackagePrefix is the default java_package prefix used in the JavaPackage modifier.
-	javaPackagePrefix = "com."
+	// defaultJavaPackagePrefix is the default java_package prefix used in the JavaPackage modifier.
+	defaultJavaPackagePrefix = "com."
```

Comment:
I'd argue we probably want to make this `com`, and define `java_package_prefix` to be used as `PREFIX.PACKAGE`, not `PREFIXPACKAGE`.

Reply:
Yea that's fair. I originally thought we should keep this open-ended so that users could customize their own prefix (e.g. they might not want a `.` after the prefix, etc.). But the gain in configuration cleanliness probably outweighs that. Plus, if there's ever a need for that kind of behavior, we can extend the configuration to support it in a backwards compatible manner.